### PR TITLE
stop the server when a client disconnects

### PR DESCRIPTION
**Pull request by jhsu**
when client disconnects (sigint or however), the server gets the close event on the websocket and will emit a `connection-closed` event which `./bin/remote.js` will listen on and close.

implications:
- websocket server will close when *any* client connection is closed

### CODE REVIEW

File: lib/ws-client.js [hunk 1]
```diff
@@ -11,11 +11,10 @@ function WSClient(params) {
     hostUrl = params.url;
     events = new EventEmitter();
     connect();
-    process.on('SIGINT', handleSigint);
 
     return {
         send: send,
-        on: events.on.bind(events)
+        on: events.on.bind(events),
```
Comment (joelgriffith) [inline lib/ws-client.js]: Stray comma?

File: lib/ws-server.js [hunk 2]
```diff
@@ -34,6 +34,9 @@ function setupServer() {
 function handleConnect(ws) {
     _ws = ws;
     ws.on('message', handleMessage);
+    ws.on('close', function() {
+        events.emit('connection-closed');
+    });
```
Comment (joelgriffith) [inline lib/ws-server.js]: Could you move this into it's own named function? Trying to steer away from anon functions as much as possible (and keep away from the CB pyramid)

File: bin/remote.js [hunk 3]
```diff
@@ -31,7 +37,7 @@ function removePath(message) {
 }
 
 server.on('file-change', function(message) {
-    if (config.debug) console.log('[' + config.hostname + '] < ' + message.changeType + ' ' + message.location);
+    if (config.debug) serverLog('< ' + message.changeType + ' ' + message.location);
```
Comment (joelgriffith) [inline bin/remote.js]: This is slick, I like me this.

File: lib/ws-client.js (hunk 1 above)
Comment (jhsu) [inline lib/ws-client.js]: nothing wrong with that =P but i'll remove it just for you

Comment (joelgriffith) [inline lib/ws-client.js]: Totally, just a style thing I guess.

**Pull request merged**